components/homme/src/theta-l/prim_advance_mod.F90

```diff
@@ -1284,7 +1284,7 @@ subroutine advance_hypervis(elem,hvcoord,hybrid,deriv,nt,nets,nete,dt2,eta_ave_w
              pnh,exner,temp_i,caller='advance_hypervis')
         
         do k=1,nlev
-           k2=max(k,nlev)
+           k2=min(k+1,nlev)
```

Comment:
Is average supposed to be
lev1+lev2, lev2+lev3,...,level nlev + level nlevp ?
I read this as 
k=1 => lev1+lev2, k=2 => lev2 + lev 3,...., k=nlev => level nlev + level min(nlev+1, nlev) = level nlev + level nlev?

Why not state k2 := k+1 ?

Reply:
the viscosity tendencies are not computed at k=nlevp. so at the surface, we cant average, and just use the nlev value.